src/server/game/Server/Packets/MailPackets.cpp

```diff
@@ -0,0 +1,93 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "MailPackets.h"
+
+void WorldPackets::Mail::MailGetList::Read()
+{
+    _worldPacket >> Mailbox;
+    LowGuids.resize(_worldPacket.read<int32>());
+
+    for (auto l : LowGuids)
+        _worldPacket >> l;
+}
+
+WorldPacket const* WorldPackets::Mail::MailListResult::Write()
+{
+    _worldPacket << int32(Mails.size());
+    _worldPacket << int32(TotalNumRecords);
+
+    for (auto const& mail : Mails)
+    {
+        _worldPacket << int32(mail.MailID);
+        _worldPacket << uint8(mail.SenderType);
+
+        _worldPacket.WriteBit(mail.SenderHint.VirtualRealmAddress.HasValue);
+        _worldPacket.WriteBit(mail.SenderHint.NativeRealmAddress.HasValue);
+
+        if (mail.SenderHint.VirtualRealmAddress.HasValue)
+            _worldPacket << int32(mail.SenderHint.VirtualRealmAddress.Value);
+
+        if (mail.SenderHint.NativeRealmAddress.HasValue)
+            _worldPacket << int32(mail.SenderHint.NativeRealmAddress.Value);
+
+        _worldPacket << int64(mail.Cod);
+        _worldPacket << int32(mail.PackageID);
+        _worldPacket << int32(mail.StationeryID);
+        _worldPacket << int64(mail.SentMoney);
+        _worldPacket << int32(mail.Flags);
+        _worldPacket << float(mail.DaysLeft);
+        _worldPacket << int32(mail.MailTemplateID);
+
+        _worldPacket << int32(mail.Attachments.size());
+
+        for (auto const& att : mail.Attachments)
+        {
+            _worldPacket << uint8(att.Position);
+            _worldPacket << uint64(att.AttachID);
+            _worldPacket << Item::ItemInstance(att.Item);
+
+            for (auto const& en : att.Enchants)
+            {
+                _worldPacket << int32(en.Enchant);
+                _worldPacket << int32(en.Duration);
+                _worldPacket << int32(en.Charges);
+            }
+
+            _worldPacket << int32(att.Count);
+            _worldPacket << int32(att.Charges);
+            _worldPacket << int32(att.MaxDurability);
+            _worldPacket << int32(att.Durability);
+            _worldPacket.WriteBit(att.Unlocked);
+        }
+
+        _worldPacket.WriteBit(mail.SenderCharacter.HasValue);
+        _worldPacket.WriteBit(mail.AltSenderID.HasValue);
+        _worldPacket.WriteBits(mail.Subject.size(), 8);
+        _worldPacket.WriteBits(mail.Body.size(), 13);
+
+        if (mail.SenderCharacter.HasValue)
+            _worldPacket << mail.SenderCharacter.Value;
+        
+        if (mail.AltSenderID.HasValue)
+            _worldPacket << mail.AltSenderID.Value;
+
+        _worldPacket.WriteString(mail.Subject);
+        _worldPacket.WriteString(mail.Body);
+    }
+    return &_worldPacket;
```

Comment:
missing flusbits (last thing written can be bits if sendercharacters, altsenderid are not defined and subject and body are empty (I doubt this can happen but better be safe than sorry)